Toolset/palettes/script editor/behaviors/revseeditorbehavior.livecodescript

```diff
@@ -5424,3 +5455,358 @@ private command EditorFieldCacheSwap pOldRuggedObjectId, pNewRuggedObjectId
+         add 1 to tCounter
+         exit repeat
+      end if
+   end repeat
+end workOnBackSlash
+
+private command foldUnfold pStart, pEnd
+   lock screen
+   set the hidden of line pStart + 1 to pEnd  of field "Script" of me to not the hidden of line pStart +1 to pEnd of field "Script" of me
+   if the hidden of line pStart +1 of field "Numbers" of me then
+      set the textStyle["underline"] of line pStart to pEnd of field "Script" of me to false
+   else
+      set the textStyle["underline"] of line pStart of field "Script" of me to true
+   end if
+   set the hidden of line pStart + 1 to pEnd  of field "Numbers" of me to not the hidden of line pStart +1 to pEnd of field "Numbers" of me
+   unlock screen
+end foldUnfold
+
+on unfoldAll
+   lock screen
+   set the hidden of line  1 to -1  of field "Script" of me to false
+   set the hidden of line 1 to - 1  of field "Numbers" of me to false
+   set the textStyle["underline"] of line 1 to - 1 of field "Script" of me to false
+   unlock screen
+end unfoldAll
+
+on foldAll
+   local tHasHidden
+   put the hidden of line 1 to - 1 of field "Script" of me into tHasHidden
+   lock screen
+   -- unfold first is some lines are hidden
+   if tHasHidden is "mixed" or tHasHidden is true then
+      unfoldAll
+   end if
+   
+   local tHandlers
+   local tScriptHTML
+   local tNumbersHTML
+   local tScriptNewHTML
+   local tNumbersNewHTML
+   local tHandlerStart
+   local tHandlerEnd
+   local tHandlerLastEnd
+   local tHidden
+   local tBeginU
+   local tEndU
+   local tScriptTemp
+   local tNumbersTemp
+   local tUnderlineText
+   
+   put " hidden" into tHidden
+   put "<u>" into tBeginU
+   put "</u>" into tEndU
+   put 0 into tHandlerLastEnd
+   
+   local tCompilationResult
+   actionCompile
+   put the result into tCompilationResult
+   if tCompilationResult <> empty then
+      then exit foldAll
+   end if
+   
+   put the cHandlers of field "Handlers" of group "Left Handler List" of group "Left Bar" into tHandlers
+   --    getHandlerList
+   --    put the result into tHandlers
+   put the htmlText of field "Script" of me into tScriptHtml
+   put the htmlText of field "Numbers" of me into tNumbersHtml
+   
+   repeat for each line aLine in tHandlers
+      put word 3 of aLine into tHandlerStart
+      put word 4 of aLine into tHandlerEnd
+      put line tHandlerLastEnd +1 to tHandlerStart - 1 of tScriptHTML & return after tScriptNewHTML
+      put line tHandlerLastEnd + 1 to tHandlerStart of tNumbersHTML & return after tNumbersNewHTML
+      put line tHandlerStart to tHandlerEnd of tScriptHTML into tScriptTemp
+      put line tHandlerStart + 1 to tHandlerEnd of tNumbersHTML into tNumbersTemp
+      put line 1 of tScriptTemp into tUnderLineText
+      
+      put char 1 to 3 of tUnderlineText & tBeginU & char 4 to - 5 of tUnderlineText & tEndU & char -4 to - 1 of tUnderlineText & cr  after tScriptNewHTML 
+      delete line 1 of tScriptTemp
+      repeat for each line aLine in tScriptTemp
```

Comment:
`aLine` here shadows `aLine` in the parent iterator. Could cause bugs down the track